core/unit_test/TestNumericTraits.hpp

```diff
@@ -147,8 +147,8 @@ struct TestNumericTraits {
   KOKKOS_FUNCTION void operator()(QuietNaN, int, int& e) const {
 #ifndef KOKKOS_COMPILER_NVHPC  // FIXME_NVHPC 23.7 nan
     using Kokkos::Experimental::quiet_NaN;
-    constexpr auto nan  = quiet_NaN<T>::value;
-    constexpr auto zero = T(0);
+    auto const nan  = quiet_NaN<T>::value;
```

Comment:
Remind me why it cannot be constexpr?

Reply:
Ah, oversight, fixed in 9d840255f84a8ba5e035f957fd8d4c855b8b7ced.

Reply:
so this was reverted to const in https://github.com/kokkos/kokkos/pull/6543/commits/4a81ff0d107c768080899e866f8fdaedbb49afbc right?

Reply:
`zero` can't be `constexpr` for `half_t` but `nan` can (and is).